src/main/java/graphql/execution/nextgen/result/ResolvedValue.java

```diff
@@ -0,0 +1,96 @@
+package graphql.execution.nextgen.result;
+
+import graphql.GraphQLError;
+import graphql.Internal;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Consumer;
+
+@Internal
+public class ResolvedValue {
+
+    private final Object completedValue;
+    private final Object localContext;
+    private final boolean nullValue;
+    private final List<GraphQLError> errors;
+
+    private ResolvedValue(Builder builder) {
```

Comment:
I am really starting to like this pattern of `builder` to the contructor.  Its a smidge easier to code future changes